Storage/Rust/RustLogins.swift

```diff
@@ -496,4 +713,67 @@ public class RustLogins {
 
         return deferred
     }
+    
+    private func migrateSQLCipherDBIfNeeded() {
+        let keychain = KeychainWrapper.sharedAppContainerKeychain
+        let rustKeys: RustLoginEncryptionKeys = RustLoginEncryptionKeys()
+        let sqlCipherLoginsKey: String? = keychain.string(forKey: rustKeys.loginsUnlockKeychainKey)
+        let sqlCipherLoginsSalt: String? = keychain.string(forKey: rustKeys.loginsSaltKeychainKey)
+        let key = try! self.getStoredKey()
+        
+        // If the sqlcipher salt or key are missing don't migrate
+        if ((sqlCipherLoginsKey ?? "").isEmpty || (sqlCipherLoginsSalt ?? "").isEmpty ) {
+            return
+        }
+        
+        defer {
+            // Delete the old key and salt regardless of if the migration succeeded.  If
+            // it failed, it's just going to fail again next time.
+            keychain.removeObject(forKey: rustKeys.loginsUnlockKeychainKey, withAccessibility: .afterFirstUnlock)
+            keychain.removeObject(forKey: rustKeys.loginsSaltKeychainKey, withAccessibility: .afterFirstUnlock)
+        }
+        let metrics = try! migrateLogins(path: self.perFieldDatabasePath, newEncryptionKey: key, sqlcipherPath: sqlCipherDatabasePath, sqlcipherKey: sqlCipherLoginsKey!, salt: sqlCipherLoginsSalt)
```

Comment:
I think the `try!` here might lead to a runtime crash if something goes wrong on the rust side, we might instead want to wrap this in a guard or even a try/catch if possible since there could be a bunch (however unlikely) of potential failures to migrate the tables on the a-s side of things.